app/controllers/application_controller.rb

```diff
@@ -8,6 +8,8 @@ def current_user
       auth_header = request.headers['HTTP_AUTHORIZATION']
       token = auth_header.split(' ').last
       @current_user ||= User.find_by(auth_token: token)
+    elsif session[:current_user_id]
+      @current_user ||= User.find_by(id: session[:current_user_id])
```

Comment:
Could we break these two methods into their own methods to make this a bit more readable.